ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/string/EncodeTest.java

```diff
@@ -41,6 +41,15 @@ public void shouldEncodeHexToAscii() {
     assertThat(udf.encode("31202b2031203d2031", "hex", "ascii"), is("1 + 1 = 1"));
     assertThat(udf.encode("ce95cebbcebbceacceb4ceb1", "hex", "ascii"), is("������������"));
     assertThat(udf.encode("c39c6265726d656e736368", "hex", "ascii"), is("��bermensch"));
+
```

Comment:
What happens if the string is `0x0x`? Or `0X`? Or `X'` without a closing quote? Are there any characters that are not allowed with these formats?

Reply:
I am assuming that you are worried about illegal hex literals (eg: 'p', 'x', '*' etc) in these formats. The function would throw an exception and we would get nulls in the end which was the default behavior before this was implemented. That is because this feature piggybacks on the pre-existing hex decoders that existed before these changes. I am adding a few more tests to demonstrate these corrupted formats.

Reply:
Also, I noticed that the pre-existing encode udf couldn't deal with Hex of odd lengths (It would throw an exception). Now, it supports odd lengths by padding a "0" to the left in the `0x...` format but would throw an exception with `X'...'` and `x'...'` formats which is consistent with https://dev.mysql.com/doc/refman/8.0/en/hexadecimal-literals.html. But I am not sure if we should support hex of odd lengths before without any of these special characters around them. It would just throw an exception like it used to

Reply:
added more tests to demonstrate that behavior